src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs

```diff
@@ -635,9 +636,47 @@ private TestMethodAttribute GetTestMethodAttribute(MethodInfo methodInfo, TestCl
         /// <returns> The <see cref="MethodInfo"/>. </returns>
         private MethodInfo GetMethodInfoForTestMethod(TestMethod testMethod, TestClassInfo testClassInfo)
         {
-            var methodsInClass = testClassInfo.ClassType.GetRuntimeMethods().ToArray();
             MethodInfo testMethodInfo;
 
+            if (testMethod.HasManagedMethodAndType)
+            {
```

Comment:
nit: Could we pull out the contents of this if block to a `GetMethodInfoUsingManagedNameHelper` to make it cleaner and easier to read.